content/testing.md

```diff
@@ -0,0 +1,683 @@
+  const testConnection = Meteor.connect(Meteor.absoluteUrl());
+
+  generateData = () => Promise.denodeify((cb) => {
+    testConnection.call('generateFixtures', cb);
+  })();
+}
+
+export { generateData };
+```
+
+Note that we've exported a client-side symbol `generateData` which is a promisified version of the method call, which makes it simpler to use this sequentially in tests.
+
+XXX: is this a good idea? 
+
+Also of note is the way we use a second DDP connection to the server in order to send these test "control" method calls.
+
+<h2 id="acceptance-testing">Acceptance testing</h2>
+
+Acceptance testing is the process of taking an unmodified version of our application and testing it from the "outside" to make sure it behaves in a way we expect. Typically if an app passes acceptance tests, we have done our job properly from a product perspective.
+
+As acceptance tests test the behavior of the application in a full browser context in a generic way, there are a range of tools that you can to specify and run such tests. In this guide we'll demonstrate using [Chimp](https://chimp.readme.io), an acceptance testing tool with a few neat Meteor-specific features that makes it easy to use.
+
+We can make Chimp a dependency of our app by installing it as an NPM development dependency:
+
+```
+npm install --save-dev chimp
+```
+
+Chimp has a variety of options for setting it up, but we can add some NPM scripts which will run the currently tests we define in Chimp's two main modes:
+
+```json
+{
+  "scripts": {
+    ...
+    "chimp-watch": "chimp --ddp=http://localhost:3000 --watch --mocha --path=tests",
+    "chimp-test": "chimp --mocha --path=tests"
+  },
+```
+[`package.json`]
+
+Chimp will now look in the `tests/` directory (otherwise ignored by the Meteor tool) for files in which you define acceptance tests. In the Todos example app, we define a simple test that ensures we can click the "create list" button:
+
+```js
+/* eslint-env mocha */
+
+// These are Chimp globals
+/* globals browser assert server */
+
+const countLists = () => {
+  browser.waitForExist('.list-todo');
+  const elements = browser.elements('.list-todo');
+  return elements.value.length;
+};
+
+describe('list ui', () => {
+  beforeEach(() => {
+    browser.url('http://localhost:3000');
+    server.call('generateFixtures');
+  });
+
+  it('can create a list @watch', () => {
+    const initialCount = countLists();
+
+    browser.click('.js-new-list');
+
+    assert.equal(countLists(), initialCount + 1);
+  });
+});
+```
+
+<h3 id="running-acceptance-tests">Running acceptance tests</h3>
+
+To run the acceptance test, we first start our meteor server with a special test driver, `tmeasday:acceptance-test-driver`. (You'll need to `meteor add` it to your app). 
+
+This test driver literally does nothing, but by running our app in full app test mode, we make all of our [test data creating methods](#creating-integration-test-data) available:
```

Comment:
Yeah, this was a relatively late idea--I'm not sure if I should include it, but it's definitely handle to be able to load test scaffolding when doing acceptance tests.